app/controllers/course/survey/responses_controller.rb

```diff
@@ -85,21 +86,23 @@ def handle_create_error(error)
 
   def build_response
     @response.experience_points_record.course_user = current_course_user
-    @response.build_missing_answers_and_options
+    @response.build_missing_answers
   end
 
   def load_answers
-    @answers ||= @response.answers.includes(:options, question: [:section, :options])
+    @answers ||= @response.answers.includes(:options)
   end
 
   def render_response_json
+    load_sections
     load_answers
-    render partial: 'response', locals: { response: @response, survey: @survey }
+    render partial: 'response',
+           locals: { response: @response, answers: @answers, survey: @survey }
   end
 
   def response_update_params
     params.
       require(:response).
-      permit(answers_attributes: [:id, :text_response, options_attributes: [:id, :selected]])
+      permit(answers_attributes: [:id, :text_response, question_option_ids: []])
```

Comment:
Should we guard the case when `question_option_ids` is from other questions (i.e. not the question of the answer) ?